### PR TITLE
Added documentation for resonator punchout

### PR DESCRIPTION
**Pull request by GabrielePalazzo**
Checklist:
- [ ] Reviewers confirm new code works as expected.
- [ ] Tests are passing.
- [ ] Coverage does not decrease.
- [ ] Documentation is updated.
- [ ] Compatibility with Qibo modules (Please edit this section if the current pull request is not compatible with the following branches).
    - [ ] Qibo: `master`
    - [ ] Qibolab: `main`
    - [ ] Qibolab_platforms_qrc: `main`

### CODE REVIEW

File: doc/source/protocols/resonator_punchout.rst [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+* the amplitude range and step must be chosen carefully because punchout can lead to extremely long experiments (that we would like to avoid), but large steps will inevitably confuse the plot;
+* the frequency usually shifts very little, but depending on :math:`g` and :math:`\Delta`, so it is difficult to set the frequency span and step;
+* if the scan in amplitude is linear, it will be difficult to have a clear view of all the three regimes at the same time. If it is possible, it may be worth to do a logarithmic scan.
+
+Note also that, in literature, this plot is often presented as a scan in the attenuation
+of the readout line. The effect is the same as of changing the amplitude (although it is
+by default a logarithmic scan), but the plot is reversed: at the top (high attenuation)
+we will see the low power regime and at the bottom (low attenuation) the high power
+one.
```
Comment (andrea-pasquale) [inline doc/source/protocols/resonator_punchout.rst:77]: Maybe it would be nice to show also a plot with attenuation with the runcard

**Review by andrea-pasquale — COMMENTED** [review]
Thanks @GabrielePalazzo.
Find below some comments.

File: doc/source/protocols/resonator_punchout.rst [hunk 2]
```diff
@@ -0,0 +1,134 @@
+Resonator punchout
+==================
+
+
+With the resonator spectroscopy at high power we found the bare resonator frequency
+(also called high power frequency), but this is not the frequency that will be used to perform measurements.
+We need to find the frequency in the low power regime, where the resonator is actually coupled
```
Comment (Edoardo-Pedicillo) [inline doc/source/protocols/resonator_punchout.rst:7]:
```suggestion
We need to find the frequency, where the resonator is actually coupled
```
The low power regime has a relative meaning, I suggest dropping it.

File: doc/source/protocols/resonator_punchout.rst [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+power. We can alse write down the maximum value of the peak.
+Since we are not interacting directly with the qubit, we are effectively measuring the
+amplitude of the ground state and, from now on we will not change the resonator frequency,
+so we expect to always measure this amplitude for the zero state. Here it is
+not needed to do a precise measurement, but it is nevertheless very useful to have an
+approximate value, that can later be used to check, in other experiments, that we are
+not exciting the qubit by error (if we see a change in amplitude, then maybe the qubit
+state has changed).
+
+It is also to run the punchout experiment with attenuation:
```
Comment (Edoardo-Pedicillo) [inline doc/source/protocols/resonator_punchout.rst:104]: What do you mean?

**Comment by andrea-pasquale** [comment]
@GabrielePalazzo could you please fix conflicts?

**Review by Edoardo-Pedicillo — APPROVED** [review]

**Pull request merged**